plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java

```diff
@@ -348,55 +345,27 @@ public Payload get(final String id, final String version) {
     if (blobRef == null) {
       return null;
     }
-
     final Blob blob = tx.requireBlob(blobRef);
     String contentType = asset.contentType();
 
-    return new StreamPayload(
-        new InputStreamSupplier()
-        {
-          @Nonnull
-          @Override
-          public InputStream get() throws IOException {
-            return blob.getInputStream();
-          }
-        },
-        blob.getMetrics().getContentSize(),
-        contentType
-    );
-  }
-
-  @Override
-  @Transactional
-  public DateTime getLastVerified(final String id, final String version) {
-    checkNotNull(id);
-    checkNotNull(version);
-
-    StorageTx tx = UnitOfWork.currentTransaction();
-
-    final Asset asset = findAsset(tx, id, version);
-
-    checkState(asset != null);
-
-    Date date = asset.formatAttributes().get(P_LAST_VERIFIED_DATE, Date.class);
-
-    log.debug("Content for {} {} last verified as of {}", id, version, date);
-    return new DateTime(checkNotNull(date));
+    final Content content = new Content(new BlobPayload(blob, contentType));
+    Content.extractFromAsset(asset, singletonList(HashAlgorithm.SHA512), content.getAttributes());
+    return content;
   }
 
   @Override
   @Transactional(retryOn = ONeedRetryException.class)
-  public void setLastVerified(final String id, final String version, final DateTime date) {
-    checkNotNull(date);
+  public void setCacheInfo(final String id, final String version, final CacheInfo cacheInfo) {
+    checkNotNull(cacheInfo);
 
     StorageTx tx = UnitOfWork.currentTransaction();
 
     Asset asset = findAsset(tx, id, version);
     checkState(asset != null);
 
-    final Date priorDate = asset.formatAttributes().get(P_LAST_VERIFIED_DATE, Date.class);
-    log.debug("Updating last verified date of {} {} from {} to {}", id, version, priorDate, date);
-    asset.formatAttributes().set(P_LAST_VERIFIED_DATE, date.toDate());
+    log.debug("Updating cacheInfo of {} {} to {}", id, version, cacheInfo);
```

Comment:
Minor: only nuget appears to log this, would be nice to have the same logging in raw and maven as well